src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -557,7 +557,11 @@ def postprocess(
         key = "logits" if self.type == "ctc_with_lm" else "tokens"
         stride = None
         for outputs in model_outputs:
-            items = outputs[key].numpy()
+            if self.framework == "pt" and outputs[key].dtype in (torch.bfloat16, torch.float16):
+                # To enable using fp16 and bf16
+                items = outputs[key].float().numpy()
```

Comment:
I saw in the other PR that we are also doing this for float16 even when numpy supports it. Maybe we should only do the `upscasting` for `bfloat16` ? WDYT @amyeroberts